test/skimNtuple2016_HHbtag.cpp

```diff
@@ -0,0 +1,6078 @@
+      theSmallTree.m_addJetForw5_pt_jetup    .resize(N_jecSources,-999.);
+      theSmallTree.m_addJetForw5_pt_jetdown  .resize(N_jecSources,-999.);
+      theSmallTree.m_addJetForw5_mass_jetup  .resize(N_jecSources,-999.);
+      theSmallTree.m_addJetForw5_mass_jetdown.resize(N_jecSources,-999.);
+
+      // Jets are ordered by pT in the theBigTree.jets_px collection
+      for (unsigned int iJet = 0; (iJet < theBigTree.jets_px->size ()) && (theSmallTree.m_njets < maxNjetsSaved); ++iJet)
+      {
+        //PF jet ID
+        if (theBigTree.PFjetID->at (iJet) < PFjetID_WP) continue; // 0 ; don't pass PF Jet ID; 1: tight, 2: tightLepVeto
+
+        // Skip H->bb candidates and VBF candidates
+        if (int(iJet)==bjet1idx || int(iJet)==bjet2idx || int(iJet)==VBFidx1 || int(iJet)==VBFidx2) continue;
+
+        // build the tlv
+        TLorentzVector tlv_additionalJet(theBigTree.jets_px->at(iJet), theBigTree.jets_py->at(iJet), theBigTree.jets_pz->at(iJet), theBigTree.jets_e->at(iJet));
+
+        // Kinematic selections + lepton cleaning
+        if (tlv_additionalJet.Pt() < 20.) continue ;
+        if (tlv_additionalJet.DeltaR(tlv_firstLepton)  < lepCleaningCone) continue ;
+        if (tlv_additionalJet.DeltaR(tlv_secondLepton) < lepCleaningCone) continue ;
+
+        // Apply PUjetID only to jets with pt < 50 GeV (https://twiki.cern.ch/twiki/bin/view/CMS/HiggsToTauTauWorkingLegacyRun2#Jets)
+        if (PUjetID_WP > -1) //PU jet ID WP = 2: loose
+        {
+          if ( !(CheckBit(theBigTree.jets_PUJetIDupdated_WP->at(iJet), PUjetID_WP)) && tlv_additionalJet.Pt()<50.) continue;
+        }
+
+        //std::cout << "  -IDX: " << iJet << " --> pt/eta/phi: " << tlv_additionalJet.Pt() <<" "<< tlv_additionalJet.Eta() <<" "<< tlv_additionalJet.Phi() << std::endl;
+
+        // get up/down uncertainty for this additional jet
+        pair <vector <double>, vector<double>> unc_additionalJet_updown = getJetUpDown(iJet, theBigTree);
+
+        // Central jets ( |eta| <= 2.4 ) that pass b-jet selection
+        if (TMath::Abs(tlv_additionalJet.Eta()) <= 2.4)
+        {
+          if (theSmallTree.m_addJetCentr1_pt < 0.)
+          {
+            theSmallTree.m_addJetCentr1_pt  = tlv_additionalJet.Pt();
+            theSmallTree.m_addJetCentr1_eta = tlv_additionalJet.Eta();
+            theSmallTree.m_addJetCentr1_phi = tlv_additionalJet.Phi();
+            theSmallTree.m_addJetCentr1_e   = tlv_additionalJet.E();
+            theSmallTree.m_addJetCentr1_btag_deepFlavor = theBigTree.bDeepFlavor_probb->at(iJet) + theBigTree.bDeepFlavor_probbb->at(iJet) + theBigTree.bDeepFlavor_problepb->at(iJet);
+            theSmallTree.m_addJetCentr1_CvsL            = getCvsL(theBigTree, iJet);
+            theSmallTree.m_addJetCentr1_CvsB            = getCvsB(theBigTree, iJet);
+            if (jets_and_HHbtag.find(iJet) != jets_and_HHbtag.end()) theSmallTree.m_addJetCentr1_HHbtag = jets_and_HHbtag[iJet];
```

Comment:
why this protection? this ```if``` should always be true, right?

Reply:
Yes, it's just a double protection (effect of a copy/paste). 
Since it's not harmful I will leave it there.